lib/brains/forms.py

```diff
@@ -1,5 +1,40 @@
 from django import forms
+from django.core.exceptions import ObjectDoesNotExist
+
+from lib.buyers.models import Buyer
 
 
 class BuyerForm(forms.Form):
     uuid = forms.CharField(max_length=255)
+
+
+class PaymentMethodForm(forms.Form):
+    buyer_uuid = forms.CharField(max_length=255)
+    nonce = forms.CharField(max_length=255)
+
+    def clean_buyer_uuid(self):
+        data = self.cleaned_data
+
+        try:
+            self.buyer = Buyer.objects.get(uuid=data['buyer_uuid'])
+        except ObjectDoesNotExist:
+            raise forms.ValidationError('Buyer does not exist.')
+
+        try:
+            self.braintreebuyer = self.buyer.braintreebuyer
+        except ObjectDoesNotExist:
+            raise forms.ValidationError('Braintree buyer does not exist.')
+
+        return data
+
+    @property
+    def braintree_data(self):
+        return {
+            'customer_id': str(self.braintreebuyer.braintree_id),
+            'payment_method_nonce': self.cleaned_data['nonce'],
+            'token': str(self.buyer.pk),
```

Comment:
It shouldn't be the buyer.pk, because it won't be unique, so you are right there. I was originally thinking that if i make all the token/provider_id thing match the pk of an object it would make life that bit easier and maybe there's some crazy disaster recovery plan we could do where we could match things up.

But thinking about it, it really doesn't make much sense. I'm going to change these apis to not supply a token, let braintree provide one and store that - less code on our end, less worrying about one which pk it is.